temporalio/worker/_workflow_instance.py

```diff
@@ -518,9 +523,9 @@ async def run_update() -> None:
                         f"Update handler for '{job.name}' expected but not found, and there is no dynamic handler. "
                         f"known updates: [{' '.join(known_updates)}]"
                     )
-                self._in_progress_updates[job.id] = HandlerExecution(
-                    job.name, defn.unfinished_policy, job.id
-                )
+                self._in_progress_updates[
+                    job.id
+                ].unfinished_policy = defn.unfinished_policy
```

Comment:
We are now forced to create the `_in_progress_updates` entry before this point (when we create the task), but we update the policy from the `defn` here, since `defn` is not known until handler coroutine execution time.